sound/soc/soc-pcm.c

```diff
@@ -1430,10 +1546,18 @@ static int dpcm_prune_paths(struct snd_soc_pcm_runtime *fe, int stream,
 		unsigned int i;
 
 		/* is there a valid CPU DAI widget for this BE */
-		widget = dai_get_widget(dpcm->be->cpu_dai, stream);
+		do_prune = 1;
+		for_each_rtd_cpu_dai(dpcm->be, i, dai) {
+			widget = dai_get_widget(dai, stream);
 
-		/* prune the BE if it's no longer in our active list */
-		if (widget && widget_in_list(list, widget))
+			/*
+			 * The BE is pruned only if none of the cpu_dai
+			 * widgets are in the active list.
+			 */
+			if (widget && widget_in_list(list, widget))
+				do_prune = 0;
+		}
+		if (!do_prune)
```

Comment:
> this will conflict with @lyakh 's work...

Sorry, @plbossart @lyakh Could you describe it in more detail? Which PR will be affected?

Reply:
@bardliao it's #1501 but I think it should be ok for you to proceed with your PR for now. I'm not sure what @plbossart's plan is, but as far as I understand, even after #1501 is fully reviewed and approved here, we'll first need to also get it reviewed on ALSA and virtualisation lists, so, it will take a while.